Audible: Fix authentication for new API token format

ztripez opened the pull request:
- Add compatible token refresh handling for new actor_access_token format
- Validate signing auth availability (preferred, stable auth method)
- Accept multiple authorization code parameter names in callback URL
- Improve error handling and diagnostic messages

Note: The python audible library hasn't been updated since 2024.
See https://github.com/mkb79/Audible/issues/832 for upstream issue.

Commit:
Fix Audible authentication for new API token format

- Add compatible token refresh handling for new actor_access_token format
- Validate signing auth availability (preferred, stable auth method)
- Accept multiple authorization code parameter names in callback URL
- Improve error handling and diagnostic messages

Commit:
Merge branch 'dev' into audible-auth-fix

MarvinSchenkel commented on music_assistant/providers/audible/audible_helper.py:
```diff
@@ -41,14 +42,77 @@
+    target_domain = "audible" if with_username else "amazon"
+    url = f"https://api.{target_domain}.{domain}/auth/token"
+
+    async with httpx.AsyncClient() as client:
```
There should already be an http session available to you via `self.mass.http_session`. No need to create a new one.

Commit:
Use shared http_session instead of creating new client

- Modify refresh_access_token_compat to accept http_session parameter
- Update to use aiohttp ClientSession (from mass.http_session)
- Remove unnecessary httpx.AsyncClient creation

MarvinSchenkel reviewed (approved):
Nice! 🚀
